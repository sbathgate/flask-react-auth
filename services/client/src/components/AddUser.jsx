import React from 'react';

const AddUser = (props) => {
    return (
      <form onSubmit={(event) => props.addUser(event)}>
        <div className="field">
          <label
            className="label is-large"
            htmlFor="input-username"
          >Username</label>
          <input
            name="username"
            id="input-username"
            className="input is-large"
            type="text"
            placeholder="Enter a username"
            required
            value={props.username}
            onChange={props.handleChange}
          />
        </div>
        <div className="field">
          <label
            className="label is-large"
            htmlFor="input-email"
          >Email</label>
          <input
            name="email"
            id="input-email"
            className="input is-large"
            type="email"
            placeholder="Enter an email address"
            required
            value={props.email}
            onChange={props.handleChange}
          />
        </div>
        <input
          type="submit"
          className="button is-primary is-large is-fullwidth"
          value="Submit"
        />
      </form>
    )
  };

export default AddUser;