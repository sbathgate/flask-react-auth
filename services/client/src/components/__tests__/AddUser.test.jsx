import React from "react";
import { render, cleanup } from "@testing-library/react";

import AddUser from "../AddUser";

afterEach(cleanup);

const props = {
  username: "",
  email: "",
  handleChange: () => {
    return true;
  },
  addUser: () => {
    return true;
  }
};

it("renders with default props", () => {
  const { getByLabelText, getByText } = render(<AddUser {...props} />);

  const usernameInput = getByLabelText("Username");
  expect(usernameInput).toHaveAttribute("type", "text");
  expect(usernameInput).toHaveAttribute("required");
  expect(usernameInput).not.toHaveValue();

  const emailInput = getByLabelText("Email");
  expect(emailInput).toHaveAttribute("type", "email");
  expect(emailInput).toHaveAttribute("required");
  expect(emailInput).not.toHaveValue();

  const buttonInput = getByText("Submit");
  expect(buttonInput).toHaveValue("Submit");
});

it("renders", () => {
  const { asFragment } = render(<AddUser {...props} />);
  expect(asFragment()).toMatchSnapshot();
});
