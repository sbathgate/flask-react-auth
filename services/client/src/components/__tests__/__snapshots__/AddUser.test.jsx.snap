// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders 1`] = `
<DocumentFragment>
  <form>
    <div
      class="field"
    >
      <label
        class="label is-large"
        for="input-username"
      >
        Username
      </label>
      <input
        class="input is-large"
        id="input-username"
        name="username"
        placeholder="Enter a username"
        required=""
        type="text"
        value=""
      />
    </div>
    <div
      class="field"
    >
      <label
        class="label is-large"
        for="input-email"
      >
        Email
      </label>
      <input
        class="input is-large"
        id="input-email"
        name="email"
        placeholder="Enter an email address"
        required=""
        type="email"
        value=""
      />
    </div>
    <input
      class="button is-primary is-large is-fullwidth"
      type="submit"
      value="Submit"
    />
  </form>
</DocumentFragment>
`;
